src/sam/functions/api/get_dmps/app.rb

```diff
@@ -36,31 +46,81 @@ def self.process(event:, context:)
       logger = Uc3DmpCloudwatch::Logger.new(source: SOURCE, request_id: req_id, event:, level: log_level)
 
       params = _process_params(event:)
-      return _respond(status: 400, errors: MSG_INVALID_SEARCH, event:) if params['owner_orcid'].nil? &&
-                                                                          params['owner_org_ror'].nil? &&
-                                                                          params['modification_day'].nil?
-
       _set_env(logger:)
-      logger.info(message: "DMP ID Search Criteria: #{params}") if logger.respond_to?(:debug)
+      logger&.info(message: "DMP ID Search Criteria: #{params}")
 
       # Fail if the Provenance could not be loaded
       claim = event.fetch('requestContext', {}).fetch('authorizer', {})['claims']
       provenance = Uc3DmpProvenance::Finder.from_lambda_cotext(identity: claim, logger:)
-      return _respond(status: 403, errors: Uc3DmpId::MSG_DMP_FORBIDDEN, event:) if provenance.nil?
+      return _respond(status: 403, errors: Uc3DmpId::Helper::MSG_DMP_FORBIDDEN, event:) if provenance.nil?
+
+      # Security check. If the Provenance is not allowed to see ALL orgs and the
+      #                 requested org is not in the list of their allowed orgs, then
+      #                 remove the requested org from the params
+      unless provenance.fetch('org_access_level', '').downcase != 'all'
+        rors = provenance.fetch('ror_list', []).map { |id| id.gsub('https://ror.org/', '') }
+        # Reply with a 403 if the requested Org is not in the list of approved RORs
+        if !params['org'].nil? && !rors.include?(params['org'])
+          errors = "Invalid ROR. You may make requests for the following ROR ids: #{provenance.fetch('ror_list', [])}"
+          return _respond(status: 403, errors:, event:) if provenance.nil?
```

Comment:
L65: "Do you need "if provenance.nil?" at the end of that line since you are returning the error based on the "`if !params['org'].nil? && !rors.include?(params['org'])`" condition?

Reply:
good catch! I think that was a copy/paste issue. Yes, provenance could never be nil there